gateway-service/src/main/java/org/zowe/apiml/gateway/config/ConnectionsConfig.java

```diff
@@ -416,4 +421,124 @@ public CorsUtils corsUtils() {
+
+        // Builder the instance information to be registered with eureka
+        // server
+        InstanceInfo.Builder builder = InstanceInfo.Builder.newBuilder();
+
+        String namespace = config.getNamespace();
+        if (!namespace.endsWith(".")) {
+            namespace = namespace + ".";
+        }
+        URL url;
+        try {
+            url = new URL(externalUrl);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+
+        builder
+            .setNamespace(namespace)
+            .setAppName(config.getAppname())
+            .setInstanceId(config.getInstanceId())
+            .setAppGroupName(config.getAppGroupName())
+            .setDataCenterInfo(config.getDataCenterInfo())
+            .setIPAddr(config.getIpAddress())
+            .setHostName(url.getHost())
+            .setPort(url.getPort())
+            .enablePort(InstanceInfo.PortType.UNSECURE, config.isNonSecurePortEnabled())
+            .setSecurePort(url.getPort())
+            .enablePort(InstanceInfo.PortType.SECURE, config.getSecurePortEnabled())
+            .setVIPAddress(config.getVirtualHostName())
+            .setSecureVIPAddress(config.getSecureVirtualHostName())
+            .setHomePageUrl(null, UriComponentsBuilder.fromUriString(externalUrl).path(config.getHomePageUrlPath()).toUriString())
+            .setStatusPageUrl(null, UriComponentsBuilder.fromUriString(externalUrl).path(config.getStatusPageUrlPath()).toUriString())
+            .setHealthCheckUrls(config.getHealthCheckUrlPath(), null,null)
+            .setASGName(config.getASGName());
+
+        // Start off with the STARTING state to avoid traffic
+        if (!config.isInstanceEnabledOnit()) {
+            InstanceInfo.InstanceStatus initialStatus = InstanceInfo.InstanceStatus.STARTING;
+            if (log.isInfoEnabled()) {
+                log.info("Setting initial instance status as: " + initialStatus);
+            }
+            builder.setStatus(initialStatus);
+        }
+        else {
+            if (log.isInfoEnabled()) {
+                log.info("Setting initial instance status as: " + InstanceInfo.InstanceStatus.UP
+                    + ". This may be too early for the instance to advertise itself as available. "
+                    + "You would instead want to control this via a healthcheck handler.");
+            }
+        }
+
+        // Add any user-specific metadata information
+        for (Map.Entry<String, String> mapEntry : config.getMetadataMap().entrySet()) {
+            String key = mapEntry.getKey();
+            String value = mapEntry.getValue();
+            // only add the metadata if the value is present
+            if (value != null && !value.isEmpty()) {
+                value = value.replace(UriComponentsBuilder.fromUriString(config.getHomePageUrl()).path("/").toUriString(), UriComponentsBuilder.fromUriString(externalUrl).path("/").toUriString());
```

Comment:
It is not necesary to evaluate UriComponentsBuilder each cycle. I would mode this code before the for statement.

Reply:
updated